plugins/csp-wms-overlays/src/WebMapService.cpp

```diff
@@ -0,0 +1,354 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//                               This file is part of CosmoScout VR                               //
+//      and may be used under the terms of the MIT license. See the LICENSE file for details.     //
+//                        Copyright: (c) 2020 German Aerospace Center (DLR)                       //
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "WebMapService.hpp"
+#include "WebMapException.hpp"
+#include "logger.hpp"
+
+#include "../../../src/cs-utils/filesystem.hpp"
+#include "../../../src/cs-utils/utils.hpp"
+
+#include <regex>
+
+#include <boost/filesystem.hpp>
+
+#include <curlpp/Easy.hpp>
+#include <curlpp/Options.hpp>
+
+namespace csp::wmsoverlays {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+WebMapService::WebMapService(std::string url, std::string cacheDir)
+    : mUrl(url)
+    , mCacheDir(cacheDir)
+    , mCacheFileName(std::regex_replace(mUrl, std::regex("[/:*]"), "_") + ".xml")
+    , mTitle(parseTitle())
+    , mSettings(parseSettings())
+    , mMapFormats(parseMapFormats())
+    , mRootLayer(parseRootLayer()) {
+  mRootLayer.getRequestableLayers(mRequestableLayers);
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+std::string WebMapService::getUrl() const {
+  return mUrl;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+std::string WebMapService::getTitle() const {
+  return mTitle;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+WebMapService::Settings WebMapService::getSettings() const {
+  return mSettings;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+WebMapLayer WebMapService::getRootLayer() const {
+  return mRootLayer;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+std::vector<WebMapLayer> WebMapService::getLayers() const {
+  return mRequestableLayers;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+std::optional<WebMapLayer> WebMapService::getLayer(std::string name) const {
+  std::vector<WebMapLayer> layers = getLayers();
+  auto                     layer  = std::find_if(
+      layers.begin(), layers.end(), [&name](WebMapLayer l) { return l.getName() == name; });
```

Comment:
Once `name` is passed as `const&` you can omit the `&` here:
```suggestion
      layers.begin(), layers.end(), [name](WebMapLayer l) { return l.getName() == name; });
```